crates/ruff_linter/src/rules/flake8_pyi/rules/duplicate_union_member.rs

```diff
@@ -27,63 +31,176 @@ use crate::checkers::ast::Checker;
 /// foo: str
 /// ```
 ///
+/// ## Fix safety
+/// This rule's fix is marked as safe unless the union contains comments.
```

Comment:
The motivation is that comments might cause information loss (e.g. link to an issue, why a certain annotation was chosen) and the developer should be able to review this as possible. Marking comments as unsafe, users are able to automatically fix all other cases (comments inside type annotations are rare), and carefully review unsafe fixes. 

Syntactically all is fine. 

If there was a decision to mark comments as safe and instead document this in the rule, I can change it to that. If so, shall we then draw that as a conclusion on https://github.com/astral-sh/ruff/issues/9790 and document it?

Reply:
In the other thread I see the suggestion to not offer a fix when comments are present, shall I continue in that direction?

Reply:
@MichaReiser I recall I saw this as tweaks one previous PR, so that's why I took it as policy:
https://github.com/astral-sh/ruff/pull/14008/commits/f099b0d371e25f4c87ae881309bf853931b18fc1

Reply:
>In the other thread I see the suggestion to not offer a fix when comments are present, shall I continue in that direction?

Yes. I spent some more time in that review to see what we do in other places and not offering a fix is the most common. I suggest that we align the behavior here and revisit the decision holistically if we see need for it.

Reply:
It's good that you waited with changing the applicability. We discussed it internally thanks to you and we agree that we should mark such rules as unsafe. See https://github.com/astral-sh/ruff/pull/14300